test/server/cards/02_SHD/upgrades/PublicEnemy.spec.ts

```diff
@@ -0,0 +1,52 @@
+describe('Public Enemy', function() {
+    integration(function(contextRef) {
+        describe('Public Enemy\'s Bounty ability', function() {
+            it('should give a unit a shield', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: [
+                            { card: 'battlefield-marine' },
+                            { card: 'atst' }
+                        ]
+                    },
+                    player2: {
+                        groundArena: [
+                            { card: 'wampa', upgrades: ['public-enemy'] },
+                            { card: 'republic-tactical-officer' }
+                        ]
+                    }
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.atst);
+                expect(context.player1).toBeAbleToSelectExactly([context.p2Base, context.wampa, context.republicTacticalOfficer]);
+                // Destroys Wampa
+                context.player1.clickCard(context.wampa);
+                expect(context.atst.damage).toBe(4);
+                // Checking Wampa is destroyed
+                expect(context.wampa).toBeInLocation('discard');
+                // Selecting Unit to give shield
```

Comment:
The comments are good, do you mind also adding a little whitespace to break up the code block for readability